bool bias_state
bool buffer_state